scripts/generate-upstream-patch.sh

```diff
@@ -56,116 +51,148 @@ generate_upstream_patch() {
+    echo "    - primitives/(common, sidechain, teeracle, teerex)"
+}
+
+function apply_woker_tips() {
+	echo "======================================================================="
+	echo "upstream_commit(s) are updated."
+	echo "upstream.patch(s) are generated."
+	echo "To apply it, RUN FROM $ROOTDIR:"
+    echo "  git am -3 --exclude=tee-worker/Cargo.lock --exclude=tee-worker/enclave-runtime/Cargo.lock --directory=tee-worker < tee-worker/upstream.patch"
+
 	echo ""
 	echo "after that, please:"
 	echo "- pay special attention: "
-	if [ "$HAS_PALLETS" == "true" ]
-	then
-		echo "  * ALL changes/conflicts from pallets/upstream.patch should ONLY apply into:"
-		echo "    - pallets/(parentchain, sidechain, teeracle, teerex, test-utils)"
-		echo "    - primitives/(common, sidechain, teeracle, teerex)"
-	fi
-	if [ "$HAS_WORKER" == "true" ]
-	then
-		echo "  * ALL changes/conflicts from tee-worker/upstream.patch patch should ONLY apply into:"
-		echo "    - tee-worker"
-	fi
-	echo "- resolve any conflicts"
+    echo "  * ALL changes/conflicts from tee-worker/upstream.patch patch should ONLY apply into:"
+    echo "    - tee-worker"
+
+    echo "- resolve any conflicts"
 	echo "- optionally update Cargo.lock file"
-	if [ "$HAS_WORKER" == "true" ]
-	then
-		echo "- apply the changes to $ROOTDIR/.github/workflows/tee-worker-ci.yml"
-	fi
-	echo "======================================================================="
-	if [ "$HAS_PALLETS" == "true" ] && [ "$HAS_WORKER" == "true" ]
-	then
-		echo "***********************************************************************"
-		echo "It is HIGHLY RECOMMENDED to apply patch and commit separately."
-		echo "If trapped in git am session, don't panic. Just resolve any conflicts and commit as usual."
-		echo "And abort the am session at the end: git am --abort"
-		echo "***********************************************************************"
-	fi
+    echo "- apply the changes to $ROOTDIR/.github/workflows/tee-worker-ci.yml"
+}
+
+OPT="$1"
+case "$OPT" in
+    p)
+        has_pallets=true ;;
+    w)
+        has_worker=true ;;
+    *)
+        usage; exit 1 ;;
+esac
+
+if [ -z "$2" ]; then
+    usage; exit 1
+fi
+NEW_COMMIT=$2
+
+HAS_PALLETS=${has_pallets:-false}
+HAS_WORKER=${has_worker:-false}
+
+if [ $HAS_PALLETS == "true" ] && [ $HAS_WORKER == "true" ]
+then
+    echo "***********************************************************************"
+    echo "It is HIGHLY RECOMMENDED to apply patch and commit separately."
+    echo "If trapped in git am session, don't panic. Just resolve any conflicts and commit as usual."
+    echo "And abort the am session at the end: git am --abort"
+    echo "***********************************************************************"
+
+    exit 1
 fi
+
+if [[ $HAS_PALLETS == "true" ]]
+then
+	generate_pallets_patch "$@"
+    apply_pallets_tips
+fi
+
+if [[ $HAS_WORKER == "true" ]]
+then
+    generate_worker_patch "$@"
+    apply_woker_tips
+fi
```

Comment:
From this logic it seems that `HAS_PALLETS` and `HAS_WORKER` can't be true simultaneously.
Why not (if I'm not mistaken):
```
case "$1" in
    p)
        generate_pallets_patch "$@"
        apply_pallets_tips
        ;;
    w)
        generate_pallets_patch "$@"
        apply_woker_tips
        ;;
    *)
        usage; exit 1 ;;
esac